sotopia/envs/evaluators.py

```diff
@@ -268,7 +287,8 @@ async def __acall__(
         response_format_class = (
             EnvResponsePlus if self.response_format == "plus" else EnvResponse
         )
-
+        if self.response_format == "goal_only":
```

Comment:
I just found this line:
https://github.com/sotopia-lab/sotopia/blob/269b4f210e4f7e77cf10a8c15ab7e4c98b48ff2e/sotopia/envs/evaluators.py#L249

I don't think is a good design, could we pass in a evaluation response class to the evaluator instead of a string? e.g. `ReachGoalLLMEvaluator(..., response_class: type[BaseClass]="EnvResponse")`? This would avoid the if in this line.

Reply:
@XuhuiZhou

Reply:
Yes! I think this is good. @bugsz wrote this line, so maybe @bugsz can fix this?

Reply:
Yes, that sounds good!

Reply:
Okay what about we first merge this and create a new pr to address this (?

Reply:
Sure, please open another issue then.